superset-frontend/packages/superset-ui-core/src/query/getClientErrorObject.ts

```diff
@@ -110,17 +110,17 @@ export function getErrorFromStatusCode(status: number): string | null {
 
 export function retrieveErrorMessage(
   str: string,
-  response: Response | JsonObject,
+  errorObject: JsonObject,
 ): string {
   const statusError =
-    'status' in response ? getErrorFromStatusCode(response.status) : null;
+    'status' in errorObject ? getErrorFromStatusCode(errorObject.status) : null;
 
-  // Prefer status code message over the response text
+  // Prefer status code message over the response or HTML text
   return statusError || parseStringResponse(str);
 }
 
-export function parseErrorJson(responseObject: JsonObject): ClientErrorObject {
-  let error = { ...responseObject };
+export function parseErrorJson(responseJson: JsonObject): ClientErrorObject {
```

Comment:
I had to lookup `ClientErrorObject` and think it could be renamed `SupersetClientErrorResponse` for clarity as it appears to be pretty much a `SupersetClientResponse` object with a set of error-related keys. It's probably outside the scope of this PR though. Just a note, feel free to address or leave for another PR

Reply:
Similarly the name `parseErrorJson` isn't great. What are we doing in this function really? Looks like we're massaging the `SupersetClientResponse` into fitting `SupersetClientErrorResponse`, kinda. `processErrorResponse` (?) Maybe GPT would have good naming suggestion, it's not great at everything but it's good at that.